encord/metadata_schema.py

```diff
@@ -0,0 +1,176 @@
+from enum import Enum
+from typing import Annotated, Literal, Sequence
+
+from pydantic import BaseModel, Field, RootModel
+
+from encord.http.v2.api_client import ApiClient
+
+__all__ = ["MetadataSchema", "MetadataSchemaError", "ClientMetadataSchemaTypeEmbedding"]
+
+
+class _NumberFormat(Enum):
+    F32 = "F32"
+    F64 = "F64"
+
+
+class _ClientMetadataSchemaTypeNumber(BaseModel):
+    ty: Literal["number"] = "number"
+    format: _NumberFormat = _NumberFormat.F64
+
+
+class _ClientMetadataSchemaTypeBoolean(BaseModel):
+    ty: Literal["boolean"] = "boolean"
+
+
+class _ClientMetadataSchemaTypeVarChar(BaseModel):
+    ty: Literal["varchar"] = "varchar"
+    max_length: int = Field(8192, ge=0, le=8192)
+
+
+class _ClientMetadataSchemaTypeDateTime(BaseModel):
+    ty: Literal["datetime"] = "datetime"
+    timezone: Literal["UTC"] = "UTC"
+
+
+class _ClientMetadataSchemaTypeEnum(BaseModel):
+    ty: Literal["enum"] = "enum"
+    values: Sequence[str] = Field([], min_length=1, max_length=256)
```

Comment:
Also need to use BaseDTO field

Reply:
why?

Reply:
SDK needs to support Pydantic 1 / 2 and BaseDTO is an interface we have that to ensure that you import the right one / use only features ambient to both.
This includes an interface to Field and BaseModel for example